src/components/sidebar/Sidebar.tsx

```diff
@@ -1,50 +1,46 @@
 import { inject, observer } from 'mobx-react';
 import * as React from 'react';
-import { RouteStore } from '../../stores/routeStore';
-import RoutesEdit from './RoutesEdit';
-import LineSearch from './LineSearch';
 import * as s from './sidebar.scss';
+import { Route, RouteComponentProps } from 'react-router-dom';
+import hslLogo from '../../assets/hsl-logo.png';
+import { RouteStore } from '../../stores/routeStore';
+import { LineStore } from '../../stores/lineStore';
+import SidebarContent from './SidebarContent';
 
-interface ISidebarProps {
+interface ISidebarProps extends RouteComponentProps<any>{
     routeStore?: RouteStore;
+    lineStore?: LineStore;
 }
 
 interface ILinelistState {
     searchInput: string;
 }
 
-@inject('routeStore')
+@inject('routeStore', 'lineStore')
 @observer
 class Sidebar extends React.Component<ISidebarProps, ILinelistState> {
     constructor(props: ISidebarProps) {
         super(props);
     }
 
-    public handleHeaderClick = () => {
-        this.props.routeStore!.clearRoutes();
-    }
-
     public render(): any {
+        const handleHeaderClick = () => {
+            this.props.routeStore!.clearRoutes();
+            this.props.lineStore!.setSearchInput('');
+            this.props.lineStore!.lineSearchVisible = true;
```

Comment:
LineSearch is always visible, remove this line

Reply:
True dat